packages/app/src/cli/services/context/identifiers-extensions.ts

```diff
@@ -117,17 +135,43 @@ export async function ensureExtensionsIds(
   return ok({
     extensions: validMatches,
     extensionIds: validMatchesById,
+    extensionsNonUuidManaged,
+  })
+}
+
+async function ensureNonUuidManagedExtensionsIds(
+  remoteConfigurationRegistrations: RemoteSource[],
+  localExtensionRegistrations: LocalSource[],
+  appId: string,
+) {
+  if (!useVersionedAppConfig()) return {}
+
+  const extensionsToCreate: LocalSource[] = []
+  const validMatches: {[key: string]: string} = {}
+  localExtensionRegistrations.forEach((local) => {
+    const possibleMatch = remoteConfigurationRegistrations.find((remote) => remote.type === local.graphQLType)
+    if (possibleMatch) validMatches[local.localIdentifier] = possibleMatch.uuid
+    else extensionsToCreate.push(local)
   })
+
+  if (extensionsToCreate.length > 0) {
+    const newIdentifiers = await createExtensions(extensionsToCreate, appId, false)
+    for (const [localIdentifier, registration] of Object.entries(newIdentifiers)) {
+      validMatches[localIdentifier] = registration.uuid
+    }
+  }
```

Comment:
We can only have 1 configuration registration and the only non-uuid-managed extension is the configuration, right? In that case, we could simplify this a lot with those assumptions, forgetting about arrays and so on.